components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AuthorizationCodeDAOImpl.java

```diff
@@ -266,6 +275,12 @@ public AuthorizationCodeValidationResult validateAuthorizationCode(String consum
                             "for client id " + consumerKey, e);
                 }
                 user.setAuthenticatedSubjectIdentifier(subjectIdentifier, serviceProvider);
+                if (StringUtils.isNotEmpty(appResidentOrganizationId)) {
+                    String userOrganizationId = OAuth2ServiceComponentHolder.getInstance().getOrganizationManager()
```

Comment:
here the user's org id is always the sub org app residing org righ?
Better to resolve the user resident org correctly as this logic will fail when the app is shared

Reply:
The user resident organization is resolved from the tenant domain. Tenant domain is extracted from the database result and it is used to create the authenticated user object as well (line 267).